spyder/utils/sourcecode.py

```diff
@@ -120,12 +125,99 @@ def get_identifiers(source_code):
     valid = re.compile(r'[a-zA-Z_]')
     return [token for token in tokens if re.match(valid, token)]
 
+def path_components(path):
+    """
+    Return the individual components of a given file path
+    string (for the local operating system).
 
-if __name__ == '__main__':
-    code = 'import functools\nfunctools.partial'
-    assert get_primary_at(code, len(code)) == 'functools.partial'
-    assert set(get_identifiers(code)) == set(['import', 'functools',
-                                              'functools.partial'])
-    assert split_source(code) == ['import functools', 'functools.partial']
-    code = code.replace('\n', '\r\n')
-    assert split_source(code) == ['import functools', 'functools.partial']
+    Taken from http://stackoverflow.com/q/21498939/438386
+    """
+    components = []
+    # The loop guarantees that the returned components can be
+    # os.path.joined with the path separator and point to the same
+    # location:    
+    while True:
+        (new_path, tail) = os.path.split(path)  # Works on any platform
+        components.append(tail)        
+        if new_path == path:  # Root (including drive, on Windows) reached
+            break
+        path = new_path
+    components.append(new_path)    
+    components.reverse()  # First component first 
+    return components
+
+def differentiate_prefix(path_components0, path_components1):
+    """
+    Return the differentiated prefix of the given two iterables. 
+     
+    Taken from http://stackoverflow.com/q/21498939/438386
+    """
+    longest_prefix = []
+    root_comparison = False
+    common_elmt = None
+    for index, (elmt0, elmt1) in enumerate(zip(path_components0, path_components1)):
+        if elmt0 != elmt1:
+            if index == 2:
+                root_comparison = True
+            break
+        else:
+            common_elmt = elmt0
+        longest_prefix.append(elmt0)
+    file_name_length = len(path_components0[len(path_components0) - 1])
+    path_0 = os.path.join(*path_components0)[:-file_name_length - 1]
+    if len(longest_prefix) > 0:
+        longest_path_prefix = os.path.join(*longest_prefix)
+        longest_prefix_length = len(longest_path_prefix) + 1
+        if path_0[longest_prefix_length:] != '' and not root_comparison:
+            path_0 = path_0[longest_prefix_length:]
+        elif not root_comparison:
+            path_0 = common_elmt
+        elif sys.platform.startswith('linux') and path_0 == '':
+            path_0 = '/'
+    return path_0
+
+def get_file_title(data, index):

```

Comment:
The Python console doesn't have the same structure for tabs as the Editor, so this is not useful.

So you need to make this function to receive a filename instead of data and index, and call it differently in the Editor.